lib/improver/tests/spotdata/spotdata/test_read_input.py

```diff
@@ -0,0 +1,360 @@
+            Plugin('single_file').process([self.cube_path, self.cube_path2],
+                                          diagnostic)
+
+
+class Test_get_method_prerequisites(Test_read_input):
+    """Test the retrieval of a predefined dictionary of additional diagnostics
+    required for given method of data extraction."""
+
+    def test_known_method(self):
+        """Test functionality with an expected method."""
+
+        expected = self.additional_data
+        method = 'model_level_temperature_lapse_rate'
+        result = get_method_prerequisites(method, self.data_directory)
+
+        self.assertArrayEqual(expected.keys(), result.keys())
+        for diagnostic in expected.keys():
+            self.assertArrayEqual(expected[diagnostic][0].data,
+                                  result[diagnostic][0].data)
+
+    def test_unknown_method(self):
+        """Test functionality with an unexpected method."""
+
+        expected = None
+        method = 'not_a_valid_method'
+        result = get_method_prerequisites(method, self.data_directory)
+        self.assertArrayEqual(expected, result)
+
+
+class Test_get_additional_diagnostics(Test_read_input):
+    """Test method for loading additional diagnostic cubes into CubeLists that
+    are returned with suitable keys in a dictionary."""
+
+    def test_available_data_files(self):
+        """Test with files available."""
+
+        diagnostic_name = 'temperature_on_height_levels'
+        result = get_additional_diagnostics(diagnostic_name,
+                                            self.data_directory)
+        self.assertIsInstance(result, CubeList)
+        self.assertArrayEqual(result[0].data,
+                              self.temperature_on_height_levels.data)
+
+    def test_missing_data_files(self):
+        """Test with missing files."""
+
+        diagnostic_name = 'temperature_on_height_levels'
+        msg = 'No relevant data files found in .*'
+        with self.assertRaisesRegexp(IOError, msg):
+            get_additional_diagnostics(diagnostic_name, 'not_a_valid_path')
+
+    def test_available_data_files_with_time_extraction(self):
+        """Test with files available and an extraction of data at a given
+        time."""
+
+        diagnostic_name = 'temperature_on_height_levels'
+        result = get_additional_diagnostics(
+            diagnostic_name, self.data_directory,
+            time_extract=self.time_extract)
+
+        self.assertIsInstance(result, CubeList)
+        self.assertArrayEqual(result[0].data,
+                              self.temperature_on_height_levels[0].data)
+
+    def test_available_data_files_with_invalid_time_extraction(self):
+        """Test with files available and an attempted extraction of data at a
+        time that is not valid."""
+
+        diagnostic_name = 'temperature_on_height_levels'
+        msg = 'No diagnostics match .*'
+        time_extract = Constraint(time=PartialDateTime(2018, 01, 01, 0))
```

Comment:
Zeroes removed. The date is irrelevant as the data source is created above and will never contain this date.